scipy/stats/_continuous_distns.py

```diff
@@ -6941,6 +6941,161 @@ def _stats(self, a):
+        '''
+        The strategy for this fit method is complex, so its steps will be
+        described here first.
+
+        1.
+        First, we attempt to find a fit to the data under the assumption that
+        the shape is less than one. This can be done analytically with closed
+        form equations. If the resulting shape is less than or equal to one,
+        then this fitting has been successful.
+
+        2.
+        If fitting under the assumption that the shape was less than or equal
+        to one was not successful, we attempt to solve under the assumption
+        that the shape is equal to or greater than one. If either location or
+        scale is fixed, then the other can be determined analytically. If only
+        the shape is fixed, or no parameters are fixed, then use `root_scalar`
+        to solve a system of equations formed by setting the partial
+        derivatives of the log-likelihood function with respect to the location
+        the the scale equal to each other. Location is used as the dependent
+        variable and then the shape and scale are analytically determined.
+
+        In many cases, the use of `np.nextafter` is utilized to prevent
+        roundoff error that causes the log likelihood equation to be very
+        large.
+        '''
+        data, fshape, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                 args, kwds)
+        # ensure that any fixed parameters don't violate constraints of the
+        # distribution before continuing. The support of the distribution
+        # is `0 < (x - loc)/scale < 1`.
+        if floc is not None:
+            if not data.min() > floc:
+                raise FitDataError('powerlaw', 0, 1)
+            if fscale is not None and not data.max() <= floc + fscale:
+                raise FitDataError('powerlaw', 0, 1)
+
+        if fscale is not None:
+            if fscale <= 0:
+                raise ValueError("Negative or zero `fscale` is outside the "
+                                 "range allowed by the distribution.")
+            if fscale <= data.ptp():
+                msg = "`fscale` must be greater than the range of data."
+                raise ValueError(msg)
+
+        def get_shape(data, loc, scale):
+            # The first-order necessary condition on `shape` can be solved in
+            # closed form. It can be used whether no matter the assumption
+            # of the value of the shape.
+            return -len(data) / np.sum(np.log((data - loc)/scale))
+
+        def get_scale(data, loc):
+            # analytical solution for `scale` based on the location.
+            # It can be used whether no matter the assumption of the value of
+            # the shape.
+            return data.max() - loc
+
+        if fscale is not None and floc is not None:
+            return get_shape(data, floc, fscale), floc, fscale
+
+        if fscale is not None:
```

Comment:
```suggestion
        # 2) only scale is fixed
        if fscale is not None:
```
and describe what happens above.